### ISSUE TITLE
Nazgul: Agile Development, 4 weeks

### ISSUE THREAD

**Issue by NazgulM** [lab-brussels-1/home#226]
## [0. Developing HTML & CSS](./0-developing-html-and-css)

Practice with the skills, tools, and workflows you will need to efficiently
develop websites written with HTML & CSS.

- 🥚 **VSCode**: you can ...
  - [ ] use the _LiveServer extension_ to launch an HTML page in the browser
  - [x] use autocomplete to begin a new HTML document, create new tags, and much
        more
  - [ ] use the _HTML CSS Support_ and _IntelliSense fo CSS class names_
        extensions to make writing HTML & CSS more efficient
- 🥚 **Formatting HTML & CSS**: You can ...
  - [x] use VSCode configurations to format your code on save
  - [x] use `npm run format` to format all the code in your repository
- [x] 🥚 **Linting CSS**: You can use the `npm run lint:css` script to find
      mistakes in your CSS, and you can fix the mistakes.
- [x] 🥚 **Validating HTML**: You can use the `npm run validate:html` script to
      find mistakes in your CSS, and you can fix the mistakes.
- 🥚 **Browser Developer Tools**: You can ...
  - [x] inspect the source code of a web page using the Sources tab
  - [x] find any part of the web page in the Elements tab of your Dev Tools DOM
        inspector
  - [x] find the styles for any element in the Elements tab
  - [ ] make temporary changes in a web page from the Elements tab to see how
        they will look

## [1. Following Plans](./1-following-plans)

Practice working together in a group to build small websites in incremental
steps according to a plan. You will be given final code, a detailed plan to
rebuild it, and will need to work as a group following the processes in
[Planning and Collaborating](https://github.com/HackYourFutureBelgium/planning-and-collaborating).

- [x] 🥚 **Project Definition and Scope**: As a group, you can agree on what you
      are trying to build, and can clearly describe the project using a README
      and a repository description.
- 🥚 **Project Constraints**: As a group you can define project constraints by
  discussing your strengths, weaknesses and context. You can set these types of
  constraints:
  - [x] _External_: constraints that come from outside your group
  - [x] _Internal, Involuntary_: constraints that come from within your group,
        but your group did not decide to have.
  - [x] _Internal, Voluntary_: constraints your group decides on to improve the
        quality of your development process and final software.
- [x] 🥚 **Communication Plans**: You can work with your group mates to agree on
      a communication plan to follow during development.
- [x] 🥚 **Communicating Challenges**: During development you can discuss with
      your group when ...
  - you are blocked and need help the agreed channels of communication.
  - your schedule or availability changes
  - you are having trouble getting along with each other
- [x] 🥚 **Using a Backlog**: You can develop from a backlog of features
      organized by priority:
  - _must-have_
  - _should-have_
  - _could-have_
- [x] 🥚 **Follow a Development Strategies**: You can follow a development
      strategy to understand the code required for _must-have_ user stories.
- [x] 🥚 **Create Issues From a Dev Strategy**: You can create issues for user
      stories in your project based on the development strategy.
- 🥚 **Retrospective**: Your group can complete a retrospective to ...
  - [x] assess how well you worked as a group
  - [ ] find ways to improve in the next project
  - [ ] compare your plan to the reality of developing your website
- 🐣 **Claiming and Completing Issues**: You can claim issues from the _TODO_
  column of a project board and ...
  - [x] track your progress with the project board
  - [x] develop your code locally on a separate branch
  - [x] send a PR that passes CI checks and meets the requirements
- [x] 🐣 **Code Review**: You can both review a group mate's PR, and
      productively participate in a review of your own code.
- [x] 🐣 **Match a Design & Acceptance Criteria**: When developing your tasks,
      you can write code that matches the project's Wireframe and meets your
      issue's Acceptance Criteria.

## [2. Writing Plans](./2-writing-plans)

Practicing writing your own plans as a group. You will be given _only_ the final
code for HTML/CSS web pages, your group will need write a detailed plan for
rebuilding the website in small steps.

- [x] 🥚 **Write User Personas**: As a group you can describing your site's
      users with User Personas, and you can use the personas to plan your
      website from the user's perspective.
- [x] 🥚 **Write User Stories**: Given a working web page, you can describe it
      with user stories and acceptance criteria for each user story.
- [x] 🐣 **Write a Backlog**: After listing all the user stories for a web page,
      you can organize them into a Backlog by priority: _must-have_,
      _should-have_ and _could-have_
- [x] 🐣 **Create a Prioritized Design**: Using your backlog and the original
      site as a guide, you can create a Wireframe of the site your group will
      build. You can use different colors and comments to indicate a priority
      for each component in the Wireframe.
- [x] 🐣 **Write a Development Strategies**: You can translate your Wireframe
      and Backlog into a Development Strategy for the _must-have_ user stories.

## [3. Adapting Plans](./3-adapting-plans)

Even the best plans are not perfect, it's not possible to know everything ahead
of time! You may be able to plan everything ahead of time when the web pages you
build are very small and you have the code ahead of time, but it's a whole
different thing when you need to build a larger web page over a longer period of
time.

In this chapter you will practice full Agile Development by developing a larger,
open-ended web pages and adapting your plans as you go. Your team will need to
make an initial plan, but the plan will change! Practice having regular meetings
with your group to review the website's progress, review your plan, and make any
changes in the plan that are necessary to keep the project on schedule.

- [x] 🐣 **Adjusting the Backlog**: As a group you can agree to changes in the
      backlog when necessary.
- [x] 🐣 **Adjusting the Design**: As a group you can agree on changes to the
      design during the development process when it becomes clear your old
      design needs updating.
- [x] 🐣 **Updating the Project Board**: You can manage the project board by
      adding, removing or updating issues when necessary. This includes changing
      labels, milestones, or assignees when necessary.

[TOP](#agile-development)

---

**Comment by NazgulM** [comment]
# Week 1

## I Need Help With:

Just need more practice and learn more html and css

## What went well?

Planning folder
HTML
CSS

## What went less well?

When I had some conflicts with npm run format, @alexanderism55 helped me a lot. When I used home PC I got some errors 403, then @Dnyandeo33 send me the link to set up SSH key, after that, I haven't any problems with GitHub. That I understand every mistake, error gives a lot.

## Lessons Learned

1) Do not mess up with master branch, and should work at your own branch, do not afraid branches. 
2) Learned how to work in team, collaborating, because every person has their own idea, it is very good practice.
3) Working on one html file, hopefully haven't any conflicts.
4) Updating all md files
5) Helping each other


## Sunday Prep Work

Started to learn

**Comment by danielhalasz** [comment]
@NazgulM teamwork at its best 😃 well done!

**Comment by NazgulM** [comment]
# Week 3

## I Need Help With:

This week challenge difficult for me as understanding naming, using Bootstrap. And have some difficulties with npm run format, because after checking I have changes in paragraphs that now no stable, not fixed, trying use position: fixed or absolute, doesn't work. Continue to research

## What went well?

Our team work, like dividing work, filling md files


## What went less well?

- Need more understanding with namings and format checking

## Lessons Learned

1) Start to create from 0 the website using Bootstrap, not everytime you can use Bootstrap, but very helpful with common formula.
2) understanding some code

## Sunday Prep Work

Fixing conflicts and errors

**Comment by danielhalasz** [comment]
@NazgulM the idea was to either use Bootstrap or regular CSS, but not really to combine both at this point. So in fact, if you use Bootstrap, there is no need for naming. Positing should be done by using the appropriate Bootstrap classes, for example.

**Comment by NazgulM** [comment]
# Week 4

## I Need Help With:

I can help myself only with studying and continue to practicing.

## What went well?

Hello coaches.
Comparing with last week, this week gone very interesting, that I started to understand with Bootstrap, if last week i used more Css in Bootstrap 5, this week, i totally used only Bootstrap, it helps me a lot. And now I understand, which way I can help myself in creating websites.


## What went less well?

- I have to more learn about spacing in Bootstrap, and continue to practice with basics.
- i want to use a lot of efffects, but without any knowledge, it just makes worse. 

## Lessons Learned
- I can totally use only Bootstrap
- Don't be afraid and just go further
- No worries with conflicts
- Just play around the website



## Sunday Prep Work

Continue practicing

**Comment by NazgulM** [comment]
## ME

### What was not clear, where did you get stuck?

- These 4 weeks gone very fast. I was amazing challenge work with two teams. First two weeks had a lot of communication and knowing each other styles etc. But after we just doing all our tasks step by step through communication.

### What was clear, what did you master?
- I started to get familiar with Bootstrap, CSS and HTML, Despite the online lessons, in case of a lot of information it just can very quickly be forgotten. One way: to continue creating websites and using different ways

### Where can you still use some help?
- All problems I faced when I starting to do something new.

### Where can you help others moving forwards?

 - I can help with planning, git commands, github repo issues, can share for now with documentation.
 
---

## The Course

### What can there be more of?
   - I almost every day busy,  for now it is fine

### What can there be less of?
- Thank you coaches, all is fine

### What material were most helpful (from HYF or elsewhere)?
- FreeCodeCamp
- Traversy Media
- Youtube channels
- W3

### What HYF material was least helpful?
- All the materials were helpful. It was nice to refer all of them so that we can gain more knowledge.

### Any suggestions for future classes?
- No suggestions.

**Comment by danielhalasz** [comment]
> * i want to use a lot of efffects, but without any knowledge, it just makes worse.

yes, try to stick to the basics until you are comfortable with CSS positioning, sizing, responsiveness..
later on you can slowly get into CSS animation and other tricks..

**Comment by danielhalasz** [comment]
> * Despite the online lessons, in case of a lot of information it just can very quickly be forgotten. One way: to continue creating websites and using different ways

this is an excellent observation. when watching videos, all seem super easy, but to do it alone from zero is where it truly sinks in & with a lot of practice and trial and error 🙂
